amlip_cpp/include/amlip_cpp/node/collaborative_learning/ModelManagerReceiverNode.hpp

```diff
@@ -0,0 +1,201 @@
+     * @param data
+     * @param domain_id DomainId of the DDS DomainParticipant
+     */
+    AMLIP_CPP_DllAPI ModelManagerReceiverNode(
+            types::AmlipIdDataType id,
+            types::ModelDataType& data,
+            uint32_t domain_id);
+
+    /**
+     * @brief Construct a new ModelManagerReceiverNode Node object.
+     *
+     * @param id Id of the Participant (associated with the Node it belongs to)
+     * @param data
+     */
+    AMLIP_CPP_DllAPI ModelManagerReceiverNode(
+            types::AmlipIdDataType id,
+            types::ModelDataType& data);
+
+    /**
+     * @brief Construct a new ModelManagerReceiverNode Node object.
+     *
+     * @param name name of the Node (it is advisable to be unique, or at least representative).
+     * @param data
+     * @param domain_id DomainId of the DDS DomainParticipant
+     */
+    AMLIP_CPP_DllAPI ModelManagerReceiverNode(
+            const char* name,
+            types::ModelDataType& data,
+            uint32_t domain_id);
+
+    /**
+     * @brief Construct a new ModelManagerReceiverNode Node object.
+     *
+     * @param name name of the Node (it is advisable to be unique, or at least representative).
+     * @param data
+     */
+    AMLIP_CPP_DllAPI ModelManagerReceiverNode(
+            const char* name,
+            types::ModelDataType& data);
+
+    /**
+     * @brief Destroy the ModelManagerReceiverNode Node object
+     *
+     * If it is processing data, it stops.
+     */
+    AMLIP_CPP_DllAPI ~ModelManagerReceiverNode();
+
+    /**
+     * @brief Process model requests
+     *
+     * @throw if node is already running.
+     */
+    void start(
+            std::shared_ptr<ModelListener> listener);
+
+    /**
+     * @brief Stop processing data.
+     *
+     * If not processing data, do nothing.
+     */
+    void stop();
+
+protected:
+
+    /**
+     * @brief Routine to be processed that read messages when available and call listener functions.
+     *
+     * Function to run from a thread.
+     * It waits in \c statistics_reader_ to have messages to read.
+     *
+     * @param listener listener to call
+     */
+    void process_routine_(
+            std::shared_ptr<ModelListener> listener);
+
+    /**
+     * @brief Thread that waits for new data to be available.
+     */
+    std::thread receiving_thread_;
+
+    /**
+     * @brief Reference to the Reader that reads statistics.
+     *
+     * This is created from DDS Participant in ParentNode, and its destruction is handled by ParentNode.
+     */
+    std::shared_ptr<dds::Reader<types::ModelStatisticsDataType>> statistics_reader_;
+
+    /**
+     * @brief Reference to the RPC Client that sends request.
+     *
+     * This is created from DDS Participant in ParentNode, and its destruction is handled by ParentNode.
+     */
+    std::shared_ptr<dds::RPCClient<types::ModelDataType, types::ModelSolutionDataType>> model_reader_;
```

Comment:
I find the name a bit confusing, cause it's a writer+reader. How about model_receiver_/requester_? Same goes for model_writer_ in model manager sender.